src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -756,6 +758,25 @@ def _generate_properties(api_version, orchestrator_type, orchestrator_version, m
     return properties
 
 
+def _get_user_assigned_identity_client_id(cli_ctx, resource_id):
+    msi_client = get_msi_client(cli_ctx)
+    pattern = '/subscriptions/.*?/resourcegroups/(.*?)/providers/microsoft.managedidentity/userassignedidentities/(.*)'
+    resource_id = resource_id.lower()
+    match = re.search(pattern, resource_id)
+    if match:
+        resource_group_name = match.group(1)
+        identity_name = match.group(2)
+        try:
+            identity = msi_client.user_assigned_identities.get(resource_group_name=resource_group_name,
+                                                               resource_name=identity_name)
+        except CloudError as ex:
+            if 'was not found' in ex.message:
+                raise CLIError("Identity {} not found.".format(resource_id))
```

Comment:
Can you use [`ResourceNotFoundError`](https://github.com/Azure/azure-cli/blob/7b927fe3a01e25377571532ec22c48425bdc8a62/src/azure-cli-core/azure/cli/core/azclierror.py#L151) here?